unchained-commons/src/main/java/unchained/commons/contract/Factory.java

```diff
@@ -0,0 +1,22 @@
+package unchained.commons.contract;
+
+import unchained.commons.utils.ReflectionUtils;
+
+import java.util.function.Function;
+
+@FunctionalInterface
+public interface Factory<T> extends Function<Object[], T> {
```

Comment:
By marking your interface as `@FunctionalInterface`, you don't need ot extend `Function<?, ?>` anymore.